### ISSUE TITLE
`pyodide --version` doesn't give me the Pyodide version

### ISSUE THREAD

**Issue by hoodmane** [pyodide/pyodide#3745]
## 🐛 Bug

When I run `pyodide --version` I want to find out which version of Pyodide I am using. The Pyodide CLI version is okay, but it's not all that useful. Would like to see the Pyodide version too.
```
$ pyodide --version
Pyodide CLI Version: 0.2.3
```

**Comment by hoodmane** [comment]
```
$ pyodide config list
emscripten_version=3.1.27
python_version=3.10.2
```
also doesn't tell me, though it has enough info to figure it out.

**Comment by rth** [comment]
yeah, the CLI version is not that useful. But if you just `pip install pyodide-cli`  (or even `pyodide-build`) it's not very clear how it can know the pyodide version without downloading the `Makefile.env` (
or `xbuildenv.tar.bz2` ). IMO the pyodide version should indeed be part of `pyodide config list`.

**Comment by hoodmane** [comment]
We also really badly need to version the `.pyodide-xbuildenv`. Since in this case I have Pyodide v0.23.0 with a Pyodide v0.22.1 xbuildenv. `pyodide config list` gets the data from the xbuildenv but it could be a different version than the version of pyodide-build.

**Comment by hoodmane** [comment]
> But if you just pip install pyodide-cli (or even pyodide-build) it's not very clear how it can know the pyodide version without downloading the Makefile.env

Currently `pyodide-build` has the same version as Pyodide I think? And `pyodide-cli` depends on it? So it's possible to get the version:
```
$ python -c 'from importlib.metadata import distribution; print(distribution("pyodide-build").version)'
0.23.0
```

**Comment by rth** [comment]
It's the other way around currently: pyodide-build depends on pyodide-cli [here](https://github.com/pyodide/pyodide-cli/blob/main/setup.cfg#L23) as the idea was one might want to do something Pyodide related (e.g. run auditwheel) without wanting to build packages.  So currently pyodide-cli is not guaranteed that pyodide-build will be installed.

**Comment by hoodmane** [comment]
Well in either case we can print the `pyodide-build` version in `pyodide --version` if `pyodide-build is installed` and I think it would be a good idea since it's much more important than the cli version...

**Comment by hoodmane** [comment]
I also wonder if we shouldn't be including `Makefile.envs` in `pyodide-buld` as a data-file.

**Comment by ryanking13** [comment]
> Well in either case we can print the pyodide-build version in pyodide --version if pyodide-build is installed and I think it would be a good idea since it's much more important than the cli version...

That sounds reasonable to me.

> I also wonder if we shouldn't be including Makefile.envs in pyodide-buld as a data-file.

Yes, I was also thinking similar: moving some build configs into pyodide-build, so we can get (+ overwrite) configs without downloading xbuildenv.

**Comment by rth** [comment]
> print the pyodide-build version in pyodide --version if pyodide-build is installed 

When used just for information, that could work. But it means you cannot also rely on this for scripts etc, because you don't know what this command will return. So it feels like bugs reports waiting to happen. 

It does also raise the question how do I get the versions of pyodide-build,  auditwheel_emscripten etc installed since e.g. `pyodide auditwheel --version` is not a command that exists. So maybe a better thing for `pyodide --help` would be to print all the versions of all packages that are installed (pyodide-build etc).

> python -c 'from importlib.metadata import version; print(version("pyodide-build"))'

Actually, we maybe need to define what is the pyodide version. For me `pyodide-build` is not necessarily the pyodide distribution version. For instance, in the long terms it should be possible to release a bug fix for pyodide-build without releasing a new distribution for all packages, particularly if we are focusing on out-of-tree builds more.

> moving some build configs into pyodide-build

We need to think about this, but from a code organization point of view. `pyodide-build` is a tool that builds files under some prefix (or some xbuildenv). So potentially you could use current `pyodide-build` to build the prefix for `0.22.1` ( or if you can't that could probably be fixed). 

However, if we start putting build configuration, then the Python / distribution version starts to be strongly coupled to the `pyodide-build`  version. I get that it's more convenient.  Right now to build wheels for several versions you need to install the pyodide-build for that version of pyodide. But in the long term once it gets more stable, I guess it could be possible to always install the latest pyodide-build version and let it build wheels for different Python/pyodide versions?

I mean it's possible but it goes in the direction of more coupling between components, so we have to think carefully if that's what we want in the long term.

**Comment by ryanking13** [comment]
> So maybe a better thing for pyodide --version would be to print all the versions of all packages that are installed (pyodide-build etc).

Oh, yes. This was actually what I meant. Though the output of `pyodide --version` will still vary depending on what packages are installed.

```sh
$ pyodide --version
Pyodide CLI Version: 0.2.3
pyodide-build: {...}
auditwheel-emscripten: {...}
...
```

> I guess it could be possible to always install the latest pyodide-build version and let it build wheels for different Python/pyodide versions?

Yes, I believe that is what we all want to do in the future. So we should definitely not go in a direction that would bind pyodide version == pyodide-build version.

**Comment by hoodmane** [comment]
> it should be possible to release a bug fix for pyodide-build without releasing a new distribution for all packages, particularly if we are focusing on out-of-tree builds more.

The fact that we can't do this is indeed very annoying.

**Comment by ryanking13** [comment]
pyodide-cli now shows pyodide-build version as well.
